Final/TCPServer.h

```diff
@@ -0,0 +1,39 @@
+#ifndef TCP_SERVER_H
+#define TCP_SERVER_H
+
+#include <unistd.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <functional>
+#include <iostream>
+
+class ThreadPool; // Forward declare class dependencies in header file
+class SyncLinkedList;
+class LinkedList;
+
+class TCPServer {
```

Comment:
let's rename this `server` move it into the `dlpf` namespace under a new `net` and `tcp` subnamespaces:

```
#ifndef TCP_SERVER_H
#define TCP_SERVER_H

#include <unistd.h>
#include <string.h>
#include <sys/types.h>
#include <sys/socket.h>
#include <netinet/in.h>
#include <netdb.h>
#include <functional>
#include <iostream>

// Forward declarations
namespace dlpf {
  namespace sync {
    class linked_list;
  } // dlpf::sync

  namespace async {
    class thread_pool;
  } // dlpf::async
} // dlpf


namespace dlpf::net::tcp {
  class server {
    public:
      // Constructor
      server(size_t max_concurrency);
      // Destructor
      ~server();

      // This is a blocking call that listens for connections on the given port and
      // serves them using the provided handler. The handler should return true
      // if the connection should be closed after serving and false otherwise.
      void listen_and_serve(size_t port, std::function<bool (int)> handler);
    private:
      const int BACKLOG_SIZE = 2;
      int sock_fd;
      dlpf::sync::linked_list open_conns;
  };
} // dlpf::net::tcp

#endif // TCP_SERVER_H
```